lib/streamlit/config.py

```diff
@@ -1301,18 +1301,23 @@ def func_with_lock():
 
 
 def _validate_theme() -> None:
-    theme_opts = get_options_for_section("theme")
+    # Import logger locally to prevent circular references
+    from streamlit.logger import get_logger
+
+    LOGGER = get_logger(__name__)
 
-    reserved_theme_names = {"auto", "dark", "light"}
+    theme_opts = get_options_for_section("theme")
     theme_name = cast(str, theme_opts["name"])
-    if theme_name and theme_name.lower() in reserved_theme_names:
-        raise RuntimeError('theme.name cannot be "Auto", "Dark", or "Light".')
+
+    if theme_name and theme_name.lower() in theme.RESERVED_THEME_NAMES:
+        LOGGER.warning('theme.name cannot be "Auto", "Dark", or "Light".')
+        set_option("theme.name", "")
 
     if (
```

Comment:
Should we make this an `elif` instead? By clearing out the name, we could run into this warning too when it could just be the first check failing

Reply:
Partially addressed in the other comment (name is optional), but also the tests check that when name is unset and the theme is otherwise complete, we only get one warning.

Let me also add a test that both warns are printed when both name is invalid and the theme is missing required options for completeness.